packages/opencensus-web-instrumentation-perf/test/test-initial-load-root-span.ts

```diff
@@ -0,0 +1,304 @@
+    attributes: {},
+  },
+  {
+    timestamp: 606.9000000024971,
+    description: 'first-contentful-paint',
+    attributes: {},
+  },
+];
+
+const EXPECTED_NAV_FETCH_ATTRIBUTES: Attributes = {
+  'http.url': 'http://localhost:4200/',
+  'http.host': 'localhost:4200',
+  'http.path': '/',
+  'http.user_agent': USER_AGENT,
+  'http.method': 'GET',
+  'http.initiator_type': 'navigation',
+  'http.next_hop_protocol': 'http/1.1',
+  'http.resp_size': 4667,
+  'http.resp_encoded_body_size': 4404,
+  'http.resp_decoded_body_size': 4404,
+};
+
+const EXPECTED_NAV_FETCH_ANNOTATIONS: Annotation[] = [
+  {
+    timestamp: 25.100000002566958,
+    description: 'fetchStart',
+    attributes: {},
+  },
+  {
+    timestamp: 28.09999999954016,
+    description: 'domainLookupStart',
+    attributes: {},
+  },
+  {
+    timestamp: 28.09999999954016,
+    description: 'domainLookupEnd',
+    attributes: {},
+  },
+  {
+    timestamp: 28.09999999954016,
+    description: 'connectStart',
+    attributes: {},
+  },
+  {
+    timestamp: 28.300000001763692,
+    description: 'connectEnd',
+    attributes: {},
+  },
+  {
+    timestamp: 28.500000000349246,
+    description: 'requestStart',
+    attributes: {},
+  },
+  {
+    timestamp: 384.6000000012282,
+    description: 'responseStart',
+    attributes: {},
+  },
+  {
+    timestamp: 394.20000000245636,
+    description: 'responseEnd',
+    attributes: {},
+  },
+];
+
+const EXPECTED_RESOURCE_ATTRIBUTES: Attributes = {
+  'http.url': 'http://localhost:4200/resource',
+  'http.host': 'localhost:4200',
+  'http.path': '/resource',
+  'http.user_agent': 'Mozilla/5.0 TEST',
+  'http.next_hop_protocol': 'h2',
+  'http.method': 'GET',
+  'http.initiator_type': 'link',
+};
+const EXPECTED_RESOURCE_ANNOTATIONS: Annotation[] = [
+  {
+    'timestamp': 266.9999999925494,
+    'description': 'fetchStart',
+    'attributes': {},
+  },
+  {
+    'timestamp': 280.1000000035856,
+    'description': 'responseEnd',
+    'attributes': {},
+  },
+];
+
+describe('getInitialLoadRootSpan', () => {
+  beforeEach(() => {
+    spyOnProperty(navigator, 'userAgent').and.returnValue(USER_AGENT);
+  });
+
+  it('creates a parent span for overall load and child spans', () => {
+    const navigationFetchSpanId = '000000000000000a';
+    const traceId = '0000000000000000000000000000000b';
+
+    const root = getInitialLoadRootSpan(
+        new Tracer(), PERF_ENTRIES, navigationFetchSpanId, traceId);
+
+    expect(root.name).toBe('Nav./');
+    expect(root.kind).toBe(SpanKind.UNSPECIFIED);
+    expect(root.parentSpanId).toBe('');
+    expect(root.id).toMatch(SPAN_ID_REGEX);
+    expect(root.traceId).toBe(traceId);
+    expect(root.startPerfTime).toBe(0);
+    expect(root.endPerfTime).toBe(20985.30000000028);
+    expect(root.attributes).toEqual(EXPECTED_ROOT_ATTRIBUTES);
+    expect(root.annotations).toEqual(EXPECTED_ROOT_ANNOTATIONS);
+
+    expect(root.spans.length).toBe(3);
+
+    const navigationFetchSpan = root.spans[0];
```

Comment:
For unpacking multiple spans, can we use array destructuring here?

Reply:
Done, good thought.